### DISCUSSION TITLE
Rating for react-aria-components

### DISCUSSION THREAD

**Discussion by rostero1** [adobe/react-spectrum#4754] [General]
Will there be a rating component for react-aria-components?

**Comment by reidbarber** [comment]
You could build this with a [RadioGroup](https://react-spectrum.adobe.com/react-aria/RadioGroup.html), in fact the ARIA APG has a [radio group rating example](https://www.w3.org/WAI/ARIA/apg/patterns/radio/examples/radio-rating/). That being said, this would be a good example for us to add to the docs eventually.

**Comment by rostero1** [comment]
Thanks, @reidbarber.  I have this implemented as a slider using `useSliderState` and the Slider example.  I think the slider finding the nearest value is a little off for half star ratings and would be better implemented as `Math.ceil(value * (1.0 / step)) / (1.0 / step)`.

Also, I tried to copy `useSliderState` so I can make the adjustment, but get a lot of typescript undefined errors.  I had this problem in the past for other components.

**Reply by reidbarber** [reply]
You should be able to use `min={0}`, `max={10}`, `step={1}` for 5 starts with half stars (zero stars inclusive), and render the stars based on the value?

**Comment by rostero1** [comment]
I see `snapValueToStep` is way more complicated than I expected.  Do you think it'd be possible to allow for passing in a custom step rounding function, but still using the precision code in `snapValueToStep`

**Comment by rostero1** [comment]
@reidbarber here's my hack of a solution if you were ever curious (for seeing how random devs try to use useSlider or improving for the demo): https://stackblitz.com/edit/stackblitz-starters-p1mx8p

**Reply by reidbarber** [reply]
Awesome, looks great!
